plugins/bundle/status.go

```diff
@@ -27,30 +29,42 @@ type Status struct {
 	Errors                   []error   `json:"errors,omitempty"`
 }
 
+
+var (
+	bundleStatusPhase = promauto.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "bundle_status_phase",
+			Help: "The status of the bundle used by this instance",
+	},[]string{"phase", "status", "revision", "name"})
+)
+
 // SetActivateSuccess updates the status object to reflect a successful
 // activation.
 func (s *Status) SetActivateSuccess(revision string) {
 	s.LastSuccessfulActivation = time.Now().UTC()
 	s.ActiveRevision = revision
+	bundleStatusPhase.WithLabelValues("activate", "success", revision, s.Name).Set(1)
 }
 
 // SetDownloadSuccess updates the status object to reflect a successful
 // download.
 func (s *Status) SetDownloadSuccess() {
+	bundleStatusPhase.WithLabelValues("download", "success", "", s.Name).Set(1)
 	s.LastSuccessfulDownload = time.Now().UTC()
 }
 
 // SetError updates the status object to reflect a failure to download or
 // activate. If err is nil, the error status is cleared.
-func (s *Status) SetError(err error) {
-
+func (s *Status) SetError(err error, phase string) {
+	
 	if err == nil {
 		s.Code = ""
 		s.Message = ""
 		s.Errors = nil
 		return
 	}
 
+	bundleStatusPhase.WithLabelValues(phase, "failure", "", s.Name).Set(1)
```

Comment:
Going off the docstring for `SetError` should we clear the failure status on the gauge label if `err == nil` too?